packages/opentelemetry-resources/src/detectors/BrowserDetector.ts

```diff
@@ -14,17 +14,56 @@
  * limitations under the License.
  */
 
-import { IResource } from '../IResource';
+import { Attributes, diag } from '@opentelemetry/api';
+import {
+  SEMRESATTRS_PROCESS_RUNTIME_DESCRIPTION,
+  SEMRESATTRS_PROCESS_RUNTIME_NAME,
+  SEMRESATTRS_PROCESS_RUNTIME_VERSION,
+} from '@opentelemetry/semantic-conventions';
 import { ResourceDetectionConfig } from '../config';
-import { Detector } from '../types';
-import { browserDetectorSync } from './BrowserDetectorSync';
+import { ResourceDetector } from '../types';
 
 /**
  * BrowserDetector will be used to detect the resources related to browser.
  */
-class BrowserDetector implements Detector {
-  detect(config?: ResourceDetectionConfig): Promise<IResource> {
-    return Promise.resolve(browserDetectorSync.detect(config));
+class BrowserDetector implements ResourceDetector {
+  detect(config?: ResourceDetectionConfig) {
```

Comment:
`: DetectedResource` ...if we want to be consistent about public interface return types, I'd leave a suggestion but this requires adding the import too